### ISSUE TITLE
NVIDIA support with docker driver

### ISSUE THREAD

**Issue by chrisroat** [kubernetes/minikube#10229]
The [doc page on using NVIDIA GPUs](https://minikube.sigs.k8s.io/docs/tutorials/nvidia_gpu/) covers the none and kvm2 drivers.  The none driver is [considered advanced](https://minikube.sigs.k8s.io/docs/drivers/none/#overview), while the kvm2 [requires a spare GPU](https://minikube.sigs.k8s.io/docs/tutorials/nvidia_gpu/#using-the-kvm2-driver).   

Is support for the docker driver feasible/planned?

**Comment by afbjorklund** [comment]
It should be _possible_, as long as the outer privileged container is launched with nvidia-docker: https://github.com/NVIDIA/nvidia-docker/issues/375

But it is not something that would work out-of-the-box. If it was a simple option (like `--runtime`), it could be considered.

Running on Linux, that is.

Not for the Docker VM...

**Comment by chrisroat** [comment]
Yes, I was considering running on Linux and I am using nvidia-docker.

I'm not overly familiar with minikube and how to adjust the launch of the "outer privileged container".  My use case is a helm chart which I run on GKE w/GPUs, which I want to iterate locally on.

**Comment by pythonwood** [comment]
nvidia + docker is easy now.  [install-guide](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html)

```
docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi
Sat Sep 18 04:14:32 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.63.01    Driver Version: 470.63.01    CUDA Version: 11.4     |
...
```

It is so good if I can use gpu in minikube

**Comment by sharifelgamal** [comment]
We don't currently have the bandwidth to implement a feature like this, but I would be happy to review a PR that does this.

**Comment by d4l3k** [comment]
I've added the beginnings of support at https://github.com/kubernetes/minikube/pull/15927. It works E2E but has the caveat that you need a custom kicbase image with an exactly matching NVML version. Reviews welcome :)

**Comment by anthonyalayo** [comment]
Thanks @d4l3k ! I think a lot of people will appreciate this. What is left for support?

**Comment by d4l3k** [comment]
@anthonyalayo need a maintainer to chime in on approach and then update the PR to better handle the Nvidia dependencies

**Comment by anthonyalayo** [comment]
Could a maintainer comment here?
